utils/utilitity.py

```diff
@@ -0,0 +1,240 @@
+                        status=consul_status))
+        return processes
+    except (ConsulException, HTTPError, RequestException) as e:
+        raise HAConsistencyException('Could not access Consul services')\
+            from e
+
+
+def is_localhost(hostname: str) -> bool:
+    name = gethostname()
+    return hostname in ('localhost', '127.0.0.1', name, f'{name}.local')
+
+
+def is_fake_leader_name(leader: str) -> bool:
+    return re.match(r'^elect[0-9]+$', leader) is not None
+
+
+def ssh_prefix(hostname: str) -> str:
+    assert hostname
+    return '' if is_localhost(hostname) else f'ssh {hostname} '
+
+
+def consul_is_active_at(hostname: str) -> bool:
+    cmd = ssh_prefix(hostname) + \
+        'sudo systemctl is-active --quiet hare-consul-agent'
+    return os.system(cmd) == 0
+
+
+def pcs_consul_is_active_at(hostname: str) -> bool:
+    cmd = ssh_prefix(hostname) + \
+        'sudo systemctl is-active --quiet hare-consul-agent*'
+    return os.system(cmd) == 0
+
+
+def exec_silent(cmd: str) -> bool:
+    return os.system(cmd) == 0
+
+
+def exec(cmd: str) -> None:
+    assert cmd
+    if exec_silent(cmd):
+        logging.info('OK')
+    else:
+        logging.error('**ERROR**')
+
+
+def process_stop(proc: Process) -> None:
+    if proc.status != 'passing':
+        return
+    label = f' ({proc.consul_name})' if proc.systemd_name.startswith('m0d@') \
+        else ''
+    logging.info(f'Stopping {proc.systemd_name}{label} at {proc.node}... ')
+    ok = exec_silent('{}sudo systemctl stop --force {}'.format(
+        ssh_prefix(proc.node), proc.systemd_name))
+    if ok:
+        logging.info(f'Stopped {proc.systemd_name}{label} at {proc.node}')
+    else:
+        logging.error(f'**ERROR** Failed to stop {proc.systemd_name}{label}'
+                      f' at {proc.node}')
+
+
+class StopProcess:
+    def __init__(self, process: Process):
+        self.process = process
+
+
+class QuitMessage:
+    pass
+
+
+class Worker(threading.Thread):
+    def __init__(self, queue: Queue):
+        super().__init__(target=self._do_work, args=(queue, ))
+
+    def _do_work(self, queue):
+        logging.debug('Started thread')
+        while True:
+            msg = queue.get()
+            if isinstance(msg, StopProcess):
+                process_stop(msg.process)
+            if isinstance(msg, QuitMessage):
+                logging.debug('Exiting thread')
+                return
+
+
+def stop_parallel(process_list: List[Process], thread_count: int = 8) -> None:
+    size = len(process_list)
+
+    if not size:
+        return
+    q: Queue = Queue(maxsize=size)
+    thread_count = min(thread_count, size)
+
+    worker_pool: List[Worker] = [Worker(q) for i in range(thread_count)]
+
+    for w in worker_pool:
+        w.start()
+
+    for proc in process_list:
+        q.put(StopProcess(process=proc))
+    for i in range(thread_count):
+        q.put(QuitMessage())
+
+    for worker in worker_pool:
+        worker.join()
+
```

Comment:
More build issues:

--> Checking files with flake8
--> Checking files with mypy
utils/hare-shutdown:23: error: Cannot find implementation or library stub for module named 'utilitity'
utils/hare-shutdown:23: note: See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports
Found 1 error in 1 file (checked 1 source file)
make[1]: *** [mypy] Error 1